src/Simplic.Icon.Service/IconService.cs

```diff
@@ -552,6 +557,22 @@ public bool SyncFilesFromDatabase()
         }
         #endregion
 
+        #region [CheckIfNameExists]
+        /// <summary>
+        /// Checks in the data base whether the given name exists.
+        /// </summary>
+        /// <returns>True if the name exists.</returns>
+        public bool CheckIfNameExists(string name)
+        {
+            var val = sqlService.OpenConnection((connection) =>
+            {
+                return connection.Query<List<int>>($"SELECT 1 FROM {TableName} WHERE name = :name", new { name = name });
```

Comment:
query will already return an IEnumerable, so no reason to await a list of int's, that would make no sense